src/pluto/operation/mask_apply.cr

```diff
@@ -0,0 +1,29 @@
+module Pluto::Operation::MaskApply
+  def apply(mask : Mask) : self
+    clone.apply!(mask)
+  end
+
+  def apply(mask : Mask, &block : (Int32, Int32, UInt8, ChannelType) -> UInt8) : self
+    clone.apply!(mask, &block)
+  end
+
+  def apply!(mask : Mask) : self
+    raise "Mask of #{mask.width}x#{mask.height} doesn't match image dimensions #{width}x#{height}" unless mask.width == width && mask.height == height
+
+    each_channel do |channel|
+      channel.map_with_index! { |pixel, i| mask.at(i) ? pixel : 0u8 }
+    end
+    self
+  end
+
+  def apply!(mask : Mask, &block : (Int32, Int32, UInt8, ChannelType) -> UInt8) : self
+    raise "Mask of #{mask.width}x#{mask.height} doesn't match image dimensions #{width}x#{height}" unless mask.width == width && mask.height == height
+
+    each_channel do |channel, channel_type|
+      channel.map_with_index! do |pixel, i|
+        mask.at(i) ? block.call(i % width, i // width, pixel, channel_type) : pixel
+      end
+    end
+    self
+  end
```

Comment:
Not sure I got these completely right - when using the blockless `apply` method, it zeros out (turns black) any pixel the mask _doesn't match_. When using the `apply` with block, it applies the block to any pixels the mask _does match_.

The blackout approach is nice if a `+` operator (or `combine` method) was added to the `Image` class that did a linear combination of pixel values. Then a mask could be used to combine two images seamlessly: `image1.apply(mask) + image2.apply(mask.invert)`.

The alter pixels that match approach is nice when you want to, well, alter those pixels in a specific way (see blue pluto spec above).

Not sure about both of those names being named the same thing though, even though the name seems to fit in both scenarios (in different ways).